bats/tests/containers/auto-start.bats

```diff
@@ -0,0 +1,118 @@
+setup() {
+    load '../helpers/load'
+}
+
+@test 'factory reset' {
+    factory_reset
+}
+
+@test 'Start up Rancher Desktop' {
+    start_application
+}
+
+@test 'Verify that initial Behavior is all set to false' {
+    run bash -c "rdctl list-settings | jq -r '.application.autoStart ' 3>&-"
+    assert_success
+    assert_output false
+    run bash -c "rdctl list-settings | jq -r '.application.startInBackground' 3>&-"
+    assert_success
+    assert_output false
+    run bash -c "rdctl list-settings | jq -r '.application.window.quitOnClose' 3>&-"
+    assert_success
+    assert_output false
+    run bash -c "rdctl list-settings | jq -r '.application.hideNotificationIcon' 3>&-"
+    assert_success
+    assert_output false
+}
+
+@test 'Enable auto start' {
+    run rdctl set --application.auto-start=true
+    assert_success
+    run bash -c "rdctl list-settings | jq -r '.application.autoStart' 3>&-"
+    assert_output true
+}
+
+@test 'Verify that the auto-start config is created' {
+
+    if is_linux; then
+        run bash -c "cat $XDG_CONFIG_HOME/autostart/rancher-desktop.desktop &>/dev/null"
+        assert_success
+    fi
+
+    if is_macos; then
+        run bash -c "cat $HOME/Library/LaunchAgents/io.rancherdesktop.autostart.plist &>/dev/null"
+        assert_success
+    fi
+
+    if is_windows; then
+        run powershell.exe -c "reg query HKCU\Software\Microsoft\Windows\CurrentVersion\Run /v RancherDesktop"
+        assert_line --index 2 --partial "Programs\Rancher Desktop\Rancher Desktop.exe"
+    fi
+}
+
+@test 'Disable auto start' {
+    run rdctl set --application.auto-start=false
+    assert_success
+    run bash -c "rdctl list-settings | jq -r '.application.autoStart' 3>&-"
+    assert_output false
+}
+
+@test 'Verify that the auto-start config is removed' {
+
+    if is_linux; then
+        run bash -c "cat $XDG_CONFIG_HOME/autostart/rancher-desktop.desktop &>/dev/null"
```

Comment:
This should be `${XDG_CONFIG_HOME:-$HOME/.config}` because maybe it's an error, but right now the `autostart` files are written into `~/.config/autostart/rancher-desktop`.

I would prefer if they got written to `~/.config/rancher-desktop/autostart`, so hold up on this test